src/web/Controller.php

```diff
@@ -261,6 +261,21 @@ public function requireLogin()
         }
     }
 
+    /**
+     * Redirects the user to the account template if they are logged in.
+     */
+    public function requireGuest()
+    {
+        $userSession = Craft::$app->getUser();
+
+        if (!$userSession->getIsGuest()) {
+            return Craft::$app
+                ->getResponse()
+                ->redirect(Craft::$app->user->returnUrl);
```

Comment:
This logic should probably live on `craft\web\User` in a `guestRequired()` method, similar to `lognRequired()`.